source/parse.h

```diff
@@ -6602,6 +6610,21 @@ class parser
             n->template_parameters = std::move(template_parameters);
         }
 
+        //  Next is an optional template specialization argument list
+        if (
+            curr() == "type"
```

Comment:
I think the two meanings of type in the declaration of classes will be confusing.

Just changing this to any other keyword works out of the box. Was there a special reason you have chosen `"type"`?

```suggestion
            curr() == "specialize"
```

Reply:
That definitely makes more sense for the case of specializing a variable.

Reply:
This reads so much better! Thank you.